### PR TITLE
1435 List commit runs

### LINKED ISSUE #1435: Commit/merge should return the run ID in case of failure
API change to support the new concept of runIDs

### PR DESCRIPTION
**Pull request by itaiad200**
closes #1435

### CODE REVIEW

File: api/controller.go [hunk 1]
```diff
@@ -410,6 +412,52 @@ func (c *Controller) GetCommitHandler() commits.GetCommitHandler {
...
+		deps.LogAction("list_commits_actions")
+		_, err = deps.Cataloger.GetCommit(deps.ctx, params.Repository, params.CommitID)
+		if errors.Is(err, db.ErrNotFound) {
+			return commits.NewListCommitRunsNotFound().WithPayload(responseError("commit not found"))
+		}
+		if err != nil {
+			return commits.NewListCommitRunsDefault(http.StatusInternalServerError).WithPayload(responseErrorFrom(err))
+		}
+
+		runsIter, err := c.deps.Actions.ListCommitRunResults(deps.ctx, params.Repository, params.CommitID)
```
Comment (nopcoder) [inline api/controller.go:439]: missing runsIter.Close()

File: api/controller.go [hunk 2]
```diff
@@ -410,6 +412,52 @@ func (c *Controller) GetCommitHandler() commits.GetCommitHandler {
...
+			},
+			{
+				Action:   permissions.ReadActionsAction,
+				Resource: permissions.RepoArn(params.Repository),
+			},
+		})
+		if err != nil {
+			return commits.NewListCommitRunsUnauthorized().WithPayload(responseErrorFrom(err))
+		}
+		deps.LogAction("list_commits_actions")
```
Comment (nopcoder) [inline api/controller.go:430]: list_commit_runs

**Review by nopcoder — CHANGES_REQUESTED** [review]

File: api/controller.go (hunk 1 above)
Comment (itaiad200) [reply api/controller.go:439]: Done.

File: api/controller.go (hunk 2 above)
Comment (itaiad200) [reply api/controller.go:430]: Done.

**Review by nopcoder — APPROVED** [review]

**Pull request merged**